tasks/swarm_token.rb

```diff
@@ -6,8 +6,8 @@
 require 'puppet'
 
 def swarm_token(node_role)
-  cmd_string = 'docker swarm join-token -q'
-  cmd_string += " #{node_role}" unless node_role.nil?
+  cmd_string = ['docker', 'swarm', 'join-token', '-q']
+  cmd_string += [" #{node_role}"] unless node_role.nil?
```

Comment:
We should try to avoid using += here.

I believe [Array#concat](https://ruby-doc.org/core-2.1.1/Array.html#method-i-concat) would be preferable in this instance.

Also lets rename the variable to cmd rather than cmd_string.

Reply:
See above.

Reply:
In this case you have changed from concatenating a string to an array so we should generally lean towards the more idiomatic approach.